Login page

TomeHirata opened the pull request.

Wolfsin committed:
create LoginPage

only view without function

Wolfsin committed:
add axios and js-sha256

Wolfsin committed:
update login Page

add Javascript
make data

Wolfsin committed:
add register Page

Wolfsin committed:
update login layout

TomeHirata commented on app/layouts/login.vue:
```diff
@@ -0,0 +1,9 @@
+<!-- Author:ZHANG CHI-->
```
login用にlayoutを分けたのには意図ありますか？

Wolfsin replied:
下のbutton、loginとregisterは要らないと思います。

TomeHirata replied:
なるほど、了解です！

TomeHirata commented on app/pages/register.vue:
```diff
@@ -0,0 +1,122 @@
+  methods: {
+    register() {
+      if (this.$refs.form.validate()) {
+        let sha256 = require("js-sha256").sha256;
```
基本varは使わないでconstかletを使ってください!

TomeHirata commented on app/pages/register.vue:
```diff
@@ -0,0 +1,122 @@
+        let sha256 = require("js-sha256").sha256;
+        var username = this.username;
+        var email = this.email;
+        var password = sha256(this.password);
```
sha256を計算するのはサーバーでやるのでフロントからは普通にそのままのパスワードを送ってください！

TomeHirata commented on app/package.json:
```diff
@@ -9,15 +9,17 @@
   },
   "dependencies": {
+    "axios": "^0.20.0",
+    "js-sha256": "^0.9.0",
```
すみません、sha256は不要です!

TomeHirata reviewed (commented):
@Wolfsin こちら確認お願いします！

Wolfsin committed:
remove Js-sha256 and update Var to let

Wolfsin commented:
別の部分更新しました。もう一度お願いします。

Wolfsin commented on app/pages/login.vue:
```diff
@@ -0,0 +1,135 @@
+          url: "http://127.0.0.1:8000/users/login",
+          data: { email: email, password: password }
+        })
+          .then(function(response) {
```
pswは間違いでも、error出ていないのて、この部分の修正お願いできますか？

TomeHirata replied:
現状パスワードは送らせてるだけでスルーしています、最終的にはエラーを返すのでcatchしてalertだけ出してもらっていいですか？

Wolfsin commented on app/pages/register.vue:
```diff
@@ -0,0 +1,121 @@
+          url: "http://127.0.0.1:8000/users/create",
+          data: {
+            username: username,
+            user_id: "string",
```
apiを参照しながら、こちら部分のパラメータはどうしますか？

TomeHirata replied:
user_idは自分でユーザーに打ち込ませて欲しいです！

Wolfsin replied:
commentも？

Wolfsin reviewed (commented):
backendの部分

Wolfsin committed:
change axios to @nuxtjs/axios

Wolfsin committed:
remove axios js

Wolfsin committed:
Update login.vue

password error message

Wolfsin committed:
create user store

See page/login.vue for specific usage

Wolfsin committed:
finished login and register

login.vue:
after login success,not jump now
register.vue:
Finished

Wolfsin committed:
create home page

Wolfsin committed:
Revert "create home page"

This reverts commit a65cb0d2e8d94ed3cbf256daf27dfde8516dddd6.

TomeHirata commented on app/store/user.js:
```diff
@@ -0,0 +1,18 @@
+// Author: ZHANG CHI
+export const state = () => ({
+  userInfo: []
```
```
ステート
userInfo: {}

ミューテーション
add(state, object){
  state.userInfo = object
}
```

actionsは不要だと思います！

TomeHirata commented on app/store/index.js:
```diff
@@ -0,0 +1,17 @@
+// Author: ZHANG CHI
+import Vue from 'vue'
```
モジュールモードで使いたいのでこのファイル消してもらってもいいですか？

Wolfsin committed:
Complete repair

del index.js
update user.js:
 json => object
 remove getData
update login.vue:
 complete userinfo store
update register.vue:
 modify declaration method

TomeHirata committed:
Merge branch 'master' into LoginPage
